### PR TITLE
[ports/keccak-tiny] New port

### PR DESCRIPTION
**Pull request by SamuelMarks**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [ ] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.
-->

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by dg0yt** [comment]
Same comments as just added for https://github.com/microsoft/vcpkg/pull/34335.

**Comment by SamuelMarks** [comment]
@dg0yt No worries; made the changes

**Comment by dg0yt** [comment]
Disclaimer: I'm a community member.

**Comment by FrankXie05** [comment]
Is this port related to `Team Keccak`?

**Comment by SamuelMarks** [comment]
@FrankXie05 No not related to that; related to this: https://en.wikipedia.org/wiki/Keccak

I'm [voluntarily] porting a project that has 4 dependencies that vcpkg has no ports for. (I'm also porting this project from bazel to CMake; and from only supporting *BSD and Linux to MSVC).

@dg0yt Oh really? - Wow after all these years you're just like me 😶‍🌫️ 
(nice work BTW)

File: ports/keccak-tiny/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,26 @@
+vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO            coruus/${PORT}
+    REF             64b6647514212b76ae7bca0dea9b7b197d1d8186
+    SHA512          5cf14061efc1b3c934dfb28a08e2a478036109c449aed41d4deb975a9f0748db06f83c1de9e5d991009d04f0220a397f5f66232a4db04bbc0dea0c624522752c
+    HEAD_REF        master
```
Comment (FrankXie05) [inline ports/keccak-tiny/portfile.cmake:8]:
```suggestion
    REPO  coruus/${PORT}
    REF  64b6647514212b76ae7bca0dea9b7b197d1d8186
    SHA512          5cf14061efc1b3c934dfb28a08e2a478036109c449aed41d4deb975a9f0748db06f83c1de9e5d991009d04f0220a397f5f66232a4db04bbc0dea0c624522752c
    HEAD_REF  master
```
Format indentation。

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

**Comment by FrankXie05** [comment]
The usage has been tested successfully locally.

**Review by FrankXie05 — DISMISSED** [review]

File: ports/keccak-tiny/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,26 @@
+vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
```
Comment (dg0yt) [inline ports/keccak-tiny/portfile.cmake:1]: This is a windows limitation.

Comment (SamuelMarks) [reply ports/keccak-tiny/portfile.cmake:1]: Ok, so what are you suggesting?

Comment (dg0yt) [reply ports/keccak-tiny/portfile.cmake:1]:
```suggestion
if(WIN32)
    vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
endif()
```

**Review by FrankXie05 — APPROVED** [review]

**Comment by dg0yt** [comment]
FTR the port specifies C90, but the packages uses `memset_s` which normally is C11.
Build fails with NDK r26 due to absence of  `memset_s`, but probably its already not usable with previous NDK releases.
Given 9 years since the last modification, let's assume android is unsupported.

**Pull request merged**